assembly/src/ast/op.rs

```diff
@@ -0,0 +1,213 @@
+use core::fmt;
+
+use super::{Block, Instruction};
+use crate::{
+    ast::AstSerdeOptions, ByteReader, ByteWriter, Deserializable, DeserializationError,
+    Serializable, SourceSpan, Span, Spanned,
+};
+
+/// Represents the Miden Assembly instruction set syntax
+///
+/// This is separate from [Instruction] in order to distinguish control flow instructions and
+/// instructions with block regions from the rest.
+#[derive(Clone)]
+#[repr(u8)]
+pub enum Op {
```

Comment:
Yes, it's short for operation, and I mostly chose that to match up with how I represent MASM in the compiler (i.e. it is called `Op` there as well, though that is largely because we _also_ have an `Instruction` type that I wanted to distinguish from).

I think it probably doesn't make a ton of sense to separate `Op` from `Instruction` in this crate - they are not in fact different things, they just have slightly different semantics than the other instructions, but that's true of any instruction set.

So I think my preference would be to either keep it as `Op`, or get rid of it entirely, but I don't think it is important to give this a clearer name, `Op` is pretty conventional for this kind of thing anyway. If anything about it is confusing, it is why we have two different types for representing instructions in MASM.